ts/src/dna_factory_v2.ts

```diff
@@ -0,0 +1,313 @@
+import {
+  Grade,
+  Rarity,
+  version,
+  DNASchemaV4,
+  DnaDataV2,
+  ParseDataPerc,
+  NeftyCodeName,
+  ParseV2,
+  DnaDataData,
+  ParseDataComputed,
+  AdvStatsJSON,
+  AdvStatsJSONValue,
+  Parse,
+} from './interfaces/types';
+import { LATEST_VERSION as LATEST_SCHEMA_VERSION } from './deps/schemas/latest';
+import { LATEST_VERSION as LATEST_ADVENTURES_STATS_VERSION } from './deps/schemas/adventures/latest';
+import dnaSchemaV4_0 from './deps/schemas/aurory_dna_v4.0.0.json';
+import { getAverageFromRaw, getLatestSubversion, randomInt, randomNormal, toPaddedHexa } from './utils';
+import { N_STATS_SOT, TACTICS_ADV_NAMES_MAP, VERSION_LENGTH } from './constants';
+import { DNAFactoryV1 } from './dna_factory_v1';
+import adventuresStatsV0_0_6 from './deps/schemas/adventures/v0.0.6.json';
+import { compressToBase64, decompressFromBase64 } from 'lz-string';
+import neftiesInfo from './deps/nefties_info.json';
+import raritiesJson from './deps/rarities.json';
+
+const dnaSchemas: Record<version, DNASchemaV4> = {
+  '4.0.0': dnaSchemaV4_0 as DNASchemaV4,
+};
+
+const adventuresStats: Record<version, AdvStatsJSON> = {
+  '0.0.6': adventuresStatsV0_0_6,
+};
+
+export class DNAFactoryV2 {
+  private latestSchemasSubversions: Record<version, version>;
+  private codeNameToKey: Record<NeftyCodeName, string>;
+  constructor() {
+    this.latestSchemasSubversions = {};
+    this.codeNameToKey = {} as Record<NeftyCodeName, string>;
+    Object.entries(this.getDNASchema(LATEST_SCHEMA_VERSION).archetypes).forEach(([key, codename]) => {
+      this.codeNameToKey[codename as NeftyCodeName] = key;
+    });
+  }
+
+  // get latest minor version from a major version
+  private getLatestSchemaSubversion(schemaVersion?: string): string {
```

Comment:
Do we still need this?

Reply:
yes to know which exact schema version has been used